src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -0,0 +1,217 @@
+ // Copyright (c) Microsoft. All rights reserved.
+ // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ 
+<%@ WebHandler Language="C#" Class="PullRequestHandler" %>
+
+// This script is called to prepare a local repo for testing
+// ASSUMES: 
+//   - Git is installed on the system to "C:\Program Files\Git\bin\git.exe"
+//   - IIS web site points to this script
+//   - This script is committed to ./src/System.Private.ServiceModel/tools/PRService
+//   - 
+//   - The Git repo is set up as follows: 
+//     [remote "origin"]
+//	       url = https://github.com/dotnet/wcf
+//	       fetch = +refs/heads/*:refs/remotes/origin/*
+//	       fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Web;
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files\Git\bin\git.exe";
+    private static string _gitRepoPath;
+    private const int _gitExecutionTimeoutMilliseconds = 100000;
+
+    public void ProcessRequest(HttpContext context)
+    {
+        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+        string prString = context.Request.QueryString["pr"];
+
+        context.Response.Write(string.Format("PR = {0}<br/>", HttpUtility.HtmlEncode(prString) ?? "Unspecified"));
+
+        StringBuilder result = new StringBuilder();
+
+        bool success = CheckPrerequisites(result);
+
+        if (success) 
+        {
+            success = CleanupMergedBranches(result); 
+        }
+
+        if (success)
+        {
+            // Check inputs to the script
+            uint pr;
+            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
+            {
+                success = SyncToPr(pr, result);
+            }
+            else
+            {
+                context.Response.Write(string.Format("Invalid PR ID specified<br/>"));
+                success = false;
+            }
+        } 
+        
+        if (!success)
+        {
+            context.Response.StatusCode = 400;
+        }
+
+        if (context.Request.IsLocal)
+        {
+            // Don't allow error results to leak to non-local machines
```

Comment:
IsLocal basically is 'am I running localhost'. So yes, IsLocal would suppress this debug spew when being called from CI. 

zhenlan had mentioned this as a potential issue as well. What are your thoughts? I kind of want to keep this output hidden mostly because I don't really want our internal implementations to be shown in its gory detail in CI output. The trade off here is that when something _does_ go wrong, it should be easy enough to go to that CI box and change pr.ashx to show this debug information as needed. 

thoughts?